frontend/src/pages/RecipientRecord/pages/CommunicationLogForm/index.js

```diff
@@ -0,0 +1,342 @@
+import pages from './pages';
+import AppLoadingContext from '../../../../AppLoadingContext';
+import {
+  updateCommunicationLogById,
+  createCommunicationLogByRecipientId,
+  getCommunicationLogById,
+} from '../../../../fetchers/communicationLog';
+
+/**
+ * this is just a simple handler to "flatten"
+ * the JSON column data into the form
+ *
+ * @param {fn} reset this is the hookForm.reset function (pass it a new set of values and it
+ *  replaces the form with those values; it also calls the standard form.reset event
+ * @param {*} event - not an HTML event, but the event object from the database, which has some
+ * information stored at the top level of the object, and some stored in a data column
+ */
+const resetFormData = (reset, updatedLog) => {
+  const {
+    data,
+    updatedAt,
+    ...fields
+  } = updatedLog;
+
+  const form = {
+    ...defaultValues,
+    ...data,
+    ...fields,
+  };
+
+  reset(form);
+};
+
+const shouldFetch = (
+  communicationLogId,
+  regionId,
+  recipientId,
+  reportFetched,
+  isAppLoading,
+  currentPage,
+) => {
+  if (
+    !currentPage
+    || !communicationLogId
+    || !regionId
+    || !recipientId
+    || communicationLogId === 'new'
+    || reportFetched
+    || isAppLoading) {
+    return false;
+  }
+  return true;
+};
+
+export default function CommunicationLogForm({ match, recipientName }) {
+  const {
+    params: {
+      recipientId,
+      regionId,
+      currentPage,
+      communicationLogId,
+    },
+  } = match;
+
+  const reportId = useRef(communicationLogId);
+
+  // for redirects if a page is not provided
+  const history = useHistory();
+
+  /* ============
+
+  * the following errors are a bit confusingly named, but
+  * I'm copying the pattern from the ActivityReport
+  */
+
+  // this error is for errors fetching reports, its the top error
+  const [error, setError] = useState();
+
+  // this is the error that appears in the sidebar
+  const [errorMessage, updateErrorMessage] = useState();
+
+  const { user } = useContext(UserContext);
+
+  /* ============ */
+
+  const [lastSaveTime, updateLastSaveTime] = useState(null);
+  const [showSavedDraft, updateShowSavedDraft] = useState(false);
+
+  // this holds the key for the date pickers to force re-render
+  // as the truss component doesn't re-render when the default value changes
+  const [datePickerKey, setDatePickerKey] = useState(`i${Date.now().toString()}`);
+
+  /* ============
+      */
+
+  const hookForm = useForm({
+    mode: 'onBlur',
+    defaultValues: {
+      ...defaultValues,
+      userId: user.id,
+      regionId,
+    },
+    shouldUnregister: false,
+  });
+
+  const formData = hookForm.getValues();
+
+  const { isAppLoading, setIsAppLoading } = useContext(AppLoadingContext);
+  const [reportFetched, setReportFetched] = useState(false);
+
+  useEffect(() => {
+    // fetch communication log data
+    async function fetchLog() {
+      if (!shouldFetch(
+        reportId.current,
```

Comment:
Should we be checking for null before accessing current?

Reply:
`reportId` will always have a `current` property (see line 84)

Reply:
jon is correct, can't be null